profile/Profile.proto

```diff
@@ -40,3 +50,40 @@ message ProfileTaskCommandQuery {
     // last commmand timestamp
     int64 lastCommandTime = 3;
 }
+
+// dumped segment snapshot
+message ProfileTaskSegemntSnapshot {
+    // profile task id
+    string taskId = 1;
+    // dumped segment id
+    UniqueId traceSegmentId = 2;
+    // dump timestamp
+    int64 time = 3;
+    // snapshot dump sequence, start with zero
+    int32 sequence = 4;
+    // snapshot stack bytes
+    bytes stack = 5;
+}
+
+message ProfileTaskSegmentStack {
+    // snapshot element list
+    repeated ProfileTaskSegemntSnapshotElement elements = 1;
+}
+
+// dumped segment snapshot element
+// such as java thread stack trace element
+message ProfileTaskSegemntSnapshotElement {
+    string className = 1;
+    string methodName = 2;
```

Comment:
We should adopt other languages about this, merging the className, method name and line number as codeSignature?